cabal-install/Distribution/Client/IndexUtils.hs

```diff
@@ -445,30 +471,32 @@ withIndexEntries index callback = do
       callback $ map toCache (catMaybes pkgsOrPrefs)
   where
     toCache :: PackageOrDep -> IndexCacheEntry
-    toCache (Pkg (NormalPackage pkgid _ _ blockNo)) = CachePackageId pkgid blockNo
+    toCache (Pkg (NormalPackage pkgid _ _ blockNo)) = CachePackageId pkgid blockNo 0 -- FIXME
     toCache (Pkg (BuildTreeRef refType _ _ _ blockNo)) = CacheBuildTreeRef refType blockNo
-    toCache (Dep d) = CachePreference d
+    toCache (Dep d) = CachePreference d 0 0 -- FIXME
```

Comment:
So for these FIXMEs, I think you just want to add the BlockNo and Timestamp to the PackageOrDep type.